data-service/src/test/java/com/amazonaws/blox/dataservice/repository/EnvironmentRepositoryDDBTest.java

```diff
@@ -198,4 +255,197 @@ public void geEnvironment() throws StorageException, EnvironmentNotFoundExceptio
+    doThrow(new AmazonServiceException(""))
+        .when(dynamoDBMapper)
+        .load(isA(EnvironmentTargetVersionDDBRecord.class));
+    environmentRepositoryDDB.getEnvironmentTargetVersion(environmentVersion.getEnvironmentId());
+  }
+
+  @Test
+  public void getEnvironmentTargetVersion() throws StorageException {
+    when(dynamoDBMapper.load(isA(EnvironmentTargetVersionDDBRecord.class)))
+        .thenReturn(environmentTargetVersionDDBRecord);
+
+    final EnvironmentVersion environmentVersionResult =
+        environmentRepositoryDDB.getEnvironmentTargetVersion(environmentVersion.getEnvironmentId());
+    assertEquals(environmentVersion, environmentVersionResult);
+
+    verify(dynamoDBMapper)
+        .load(EnvironmentTargetVersionDDBRecord.withHashKey(environmentVersion.getEnvironmentId()));
+  }
+
+  @Test(expected = StorageException.class)
+  public void listClustersStorageException() throws StorageException {
+    when(dynamoDBMapper.scan(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBScanExpression.class)))
+        .thenThrow(new AmazonServiceException(""));
+
+    environmentRepositoryDDB.listClusters();
+  }
+
+  @Test
+  public void listClusters() throws StorageException {
+    List<EnvironmentTargetVersionDDBRecord> environmentTargetVersionDDBRecords = new ArrayList<>();
+    environmentTargetVersionDDBRecords.add(environmentTargetVersionDDBRecord);
+    final EnvironmentTargetVersionDDBRecord secondRecord =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(SECOND_CLUSTER)
+            .build();
+    environmentTargetVersionDDBRecords.add(secondRecord);
+
+    final EnvironmentTargetVersionDDBRecord thirdRecordSameCluster =
+        EnvironmentTargetVersionDDBRecord.builder()
+            .environmentId(SECOND_ENVIRONMENT_ID)
+            .environmentVersion(UUID.randomUUID().toString())
+            .cluster(SECOND_CLUSTER)
+            .build();
+    environmentTargetVersionDDBRecords.add(thirdRecordSameCluster);
+
+    when(environmentTargetVersionDDBRecordPaginatedScanList.stream())
+        .thenReturn(environmentTargetVersionDDBRecords.stream());
+
+    when(dynamoDBMapper.scan(
+            eq(EnvironmentTargetVersionDDBRecord.class), isA(DynamoDBScanExpression.class)))
+        .thenReturn(environmentTargetVersionDDBRecordPaginatedScanList);
+
+    final List<String> clusters = environmentRepositoryDDB.listClusters();
+
+    final Set<String> expectedClusters = new HashSet<>();
+    for (EnvironmentTargetVersionDDBRecord record : environmentTargetVersionDDBRecords) {
+      expectedClusters.add(record.getCluster());
+    }
+
+    assertEquals(expectedClusters.size(), clusters.size());
+    for (String c : expectedClusters) {
+      if (!clusters.contains(c)) {
+        fail(String.format("Does not contain %s", c));
```

Comment:
We've got Hamcrest, so `assertThat(clusters, containsInAnyOrder(expectedClusters.toArray()))` should work.